### PR TITLE
docs: adding architecture diagrams

### PR DESCRIPTION
**Pull request by ismaelfaro**
### Which issue(s) does this pull-request address?

Improving the documentation with architecture diagrams.

### Description

Adding diagrams about the services, and key code architecture.

### Checklist

- [x] I have read the [contributor guide](https://github.com/i-am-bee/bee-agent-framework/blob/main/CONTRIBUTING.md)
- [x] Linting passes: `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: `yarn format` or `yarn format:fix`
- [ ] Unit tests pass: `yarn test:unit`
- [ ] E2E tests pass: `yarn test:e2e`
- [ ] Tests are included 
- [x] Documentation is changed or added
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: docs/diagrams/class.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        }
+    }
+
+    %% Inheritance Relationships
+    BaseAgent --|> Serializable
+    BaseLLM --|> Serializable
+    BaseMemory --|> Serializable
+
+    SlidingMemory --|> BaseMemory
+    TokenMemory --|> BaseMemory
```
Comment (Tomas2D) [inline docs/diagrams/class.md:123]: Missing `UnconstrainedMemory`

File: docs/diagrams/class.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    LangChainAdapter --|> BaseLLM
+
+    %% Associations
+    BaseAgent --> "1" BaseLLM : uses
+    BaseAgent --> "1..*" ChatLLM : uses
+    BaseAgent --> "1" BaseMemory : uses
+    BaseAgent --> "1" Logger : uses
+    BaseTool --> "0..1" BaseCache : uses
+    BaseAgent --> "1" Emitter : emits events
+    AgentManager --> "*" BaseAgent : manages
```
Comment (Tomas2D) [inline docs/diagrams/class.md:151]: This class has been removed from the framework.

File: docs/diagrams/components.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+---
+title: Components Diagrams
+slug: architecture/diagrams
+---
+
+# Bee Stack Components
```
Comment (Tomas2D) [inline docs/diagrams/components.md:6]: This diagram should be in the https://github.com/i-am-bee/bee-stack repository.

File: docs/diagrams/services.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+---
+title: Architecture Diagrams
+slug: architecture/diagrams
+---
+
+# Bee Stack Services
+
+This diagram shows all the key Services in the Bee Stack project.
```
Comment (Tomas2D) [inline docs/diagrams/services.md:8]: This diagram should be in the https://github.com/i-am-bee/bee-stack repository.

**Review by Tomas2D — CHANGES_REQUESTED** [review]
I like it. But is there a way to generate such diagrams automatically?

File: docs/diagrams/services.md (hunk 4 above)
Comment (ismaelfaro) [reply docs/diagrams/services.md:8]: let me run the script again to update the message, and I'll only keep in this repo the Class diagram.

**Comment by Tomas2D** [comment]
What is the current state? @ismaelfaro

**Review by Tomas2D — COMMENTED** [review]
Could you please update the diagrams to reflect the actual state?

**Comment by ismaelfaro** [comment]
Let me close this PR and upload a new one with a agent that generate the diagram.

**Pull request closed without merging**